python/pyspark/sql/types.py

```diff
@@ -1528,6 +1528,12 @@ def asDict(self, recursive=False):
 
         :param recursive: turns the nested Rows to dict (default: False).
 
+        NOTE: If a row contains duplicate field names, e.g., the rows of a join
+        between two :class:`DataFrame` that both have the fields of same names,
+        ``asDict`` will return the rightmost value among the duplicate fields. In
+        contrast, ``__getitem__`` will return the leftmost value among the duplicate
```

Comment:
@viirya, this is a nice description but what about saying like one of duplicate fields will be selected? I was wondering if we should fix such rightmost or leftmost order between `asDict` and `__getitem__` in the future, and if it's better to avoid saying the rightmost and leftmost.